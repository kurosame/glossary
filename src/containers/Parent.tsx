import Child from '@/components/Child'
import { addCount, getSagaCount } from '@/modules/counter'
import { IActions, IStates } from '@/modules/index'
import { getWords } from '@/modules/word'
import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators, Dispatch } from 'redux'
import styled from 'styled-components'

interface IProps {
  states: IStates
  actions: IActions
}

class Parent extends React.Component<IProps> {
  constructor(props: IProps) {
    super(props)
    props.actions.word.getWords()
  }

  public render() {
    return (
      <Div>
        Parent
        <Child
          state={this.props.states.counter}
          actions={this.props.actions.counter}
        />
      </Div>
    )
  }
}

const Div = styled.div`
  color: white;
  background-color: green;
`

export default connect(
  (states: IStates) => ({ states }),
  (dispatch: Dispatch) => ({
    actions: {
      counter: {
        addCount: bindActionCreators(addCount, dispatch),
        getSagaCount: bindActionCreators(getSagaCount, dispatch)
      },
      word: {
        getWords: bindActionCreators(getWords, dispatch)
      }
    }
  })
)(Parent)
