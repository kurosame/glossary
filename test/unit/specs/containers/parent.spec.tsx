import Parent from '@/containers/Parent'
import { shallow } from 'enzyme'
import React from 'react'
import configureStore from 'redux-mock-store'

const states = { counter: { count: 1, sagaCount: 2 } }

const wrapper = shallow(<Parent />, {
  context: { store: configureStore()(states) }
})

test('Match the snapshot', () => {
  // TODO: https://github.com/kurosame/react-boilerplate/issues/1
  expect(wrapper.html()).toMatchSnapshot()
})
