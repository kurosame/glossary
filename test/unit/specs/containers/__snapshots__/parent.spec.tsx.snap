// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot 1`] = `"<div class=\\"sc-bwzfXH kFKloX\\">Parent<div class=\\"sc-bdVaJa fQaGEY\\"><div><span class=\\"count\\">1</span><button class=\\"add-count\\">ADD</button></div><div><span class=\\"saga-count\\">2</span><button class=\\"add-saga-count\\">ADD</button>â€»redux-saga sample</div></div></div>"`;
